app/models/post.rb

```diff
@@ -1,34 +1,13 @@
 class Post < ActiveRecord::Base
-  belongs_to :user
-  has_and_belongs_to_many :tags
-  has_many :comments, as: :parent
 
+  has_and_belongs_to_many :tags
   validates_presence_of :title
-  validates_presence_of :user, :unless => :child?
 
-  before_save do
-    unless child?
-      parser = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true)
-      self.body_html = parser.render(body_markdown)
-    end
-  end
+  scope :internalpost, -> { where(type: "InternalPost") }
+  scope :externalpost, -> { where(type: "ExternalPost") }
 
-  # after_create :notify_twitter
-  # TODO: move this to background job
-  def notify_twitter
-    $twitter_client.update(tweet_content)
+  def types
```

Comment:
@FerPerales It's a way to know which posts will subclass the Post model, like: 
``Post.types``
`` => ['InternalPost', 'ExternalPost'] ``

Maybe it's not that necessary, just a 'nice to have'.